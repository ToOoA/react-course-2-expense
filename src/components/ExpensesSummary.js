import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
        <h1>
            Viewing {
                props.expenses.length
            } expense{
                props.expenses.length > 1 && 's'
            } totalling {
                numeral(expensesTotal(props.expenses) / 100).format('$0,00.00')
            }
        </h1>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);