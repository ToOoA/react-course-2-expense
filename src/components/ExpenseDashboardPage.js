import React from 'react';

import ExpenseList from './ExpenseList';
import ExpenseListFIlters from './ExpenseListFIlters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFIlters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;