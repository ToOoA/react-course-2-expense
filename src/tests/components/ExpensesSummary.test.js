import React from 'react';
import { shallow } from 'enzyme';

import expenses from '../fixtures/expenses';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly for one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]}/>);
    expect(wrapper).toMatchSnapshot();
});