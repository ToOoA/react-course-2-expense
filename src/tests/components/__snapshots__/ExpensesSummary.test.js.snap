// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly 1`] = `
<div>
  <h1>
    Viewing 
    3
     expense
    s
     totalling 
    $1,141.95
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly for one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     expense
     totalling 
    $1.95
  </h1>
</div>
`;
